backend/src/prompts/templates/agent-selection-system-prompt.j2

```diff
@@ -35,6 +35,7 @@ The user may also provide a list of ChatAgentFailure(s), as shown below:
 ChatAgentFailure(agent_name='AGENT_NAME', reason='REASON', retry=True or False)
 
 You will take into consideration whether the agent wants to be retried or not.
+Do not retry an agent more than once.
```

Comment:
It might be better to say
If an agent has been listed twice in the ChatAgentFailures it has already tried twice to solve the problem and failed do not pick this agent

however if you really do want to remove agents that have been tried twice it would be better to do that deterministically and remove them from the list of available agents